syntax = "proto3";

service EtherIP {
  rpc CreateNewEtherIPTunnelEntry (CreateNewEtherIPTunnelEntryRequest) returns (CreateNewEtherIPTunnelEntryResponse) {}

  rpc AttachEncapsProgram (AttachEncapsProgramRequest) returns (AttachEncapsProgramResponse) {}
  rpc AttachDecapsProgram (AttachDecapsProgramRequest) returns (AttachDecapsProgramResponse) {}

  rpc DetachEncapsProgram (DetachEncapsProgramRequest) returns (DetachEncapsProgramResponse) {}
  rpc DetachDecapsProgram (DetachDecapsProgramRequest) returns (DetachDecapsProgramResponse) {}
}

message CreateNewEtherIPTunnelEntryRequest {
  string remote_addr = 3;
  string local_addr = 4;
}

message CreateNewEtherIPTunnelEntryResponse {
  int32 entry_index = 2;
  CreateNewEtherIPTunnelEntryRequest request = 3; 
}

message AttachEncapsProgramRequest {
  string ifname = 4;
  int32 entry_index = 2;
}

message AttachEncapsProgramResponse {
  AttachEncapsProgramRequest request = 2;
}

message AttachDecapsProgramRequest {
  string ifname = 3;
}

message AttachDecapsProgramResponse {
  AttachDecapsProgramRequest request = 2;
}

message DetachEncapsProgramRequest {
  string ifname = 1;
}

message DetachEncapsProgramResponse {
  DetachEncapsProgramRequest request = 1;
}

message DetachDecapsProgramRequest {
  string ifname = 1;
}

message DetachDecapsProgramResponse {
  DetachDecapsProgramRequest request = 1;
}
